tst/regression/test_suites/advection_performance/advection_performance.py

```diff
@@ -33,12 +33,35 @@
 class TestCase(utils.test_case.TestCaseAbs):
     def Prepare(self, parameters, step):
 
+        # splits integer n into the three largest factors
+        def get_split(n):
+            i = 2
+            factors = []
+            while i*i <= n:
+                if n % i != 0:
+                    i += 1
+                else:
+                    n //= i
+                    factors.append(i)
+            # always append remainder, could also be 1
+            factors.append(n)
+
+            # fill to 3 dims
+            while len(factors) < 3:
+                factors.append(1)
+            # split to 3 dims
+            while len(factors) > 3:
+                factors = sorted(factors[:-2] + [factors[-2]*factors[-1]], reverse=True)
+            return sorted(factors, reverse=True)
+
+        num_proc_x, num_proc_y, num_proc_z = get_split(parameters.num_ranks)
```

Comment:
It feels like there should be a built-in for this. But on a quick google I couldn't find one. 🤷